packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -211,44 +221,77 @@ export class ResourceBtmService {
+
+    // remove duplicates and put on the microservice with the least amount of tables
+    promptResultObj.microservices = promptResultObj.microservices
+      .sort((a, b) => a.tables.length - b.tables.length)
+      .map((microservice) => ({
+        name: microservice.name,
+        functionality: microservice.functionality,
+        tables: microservice.tables.filter((tableName) => {
+          const isDuplicatedAlreadyUsed = usedDuplicatedEntities.has(tableName);
+          if (duplicatedEntities.has(tableName)) {
+            usedDuplicatedEntities.add(tableName);
+          }
+          return (
+            originalResourceEntityNamesSet.has(tableName) &&
+            !isDuplicatedAlreadyUsed
+          );
+        }),
+      }))
+      .filter((microservice) => microservice.tables.length > 0);
+
+    // add the original resource to the microservices
+    promptResultObj.microservices.push({
+      name: `${originalResource.name}Original`,
+      functionality: originalResource.description,
+      tables: originalResourceTables.map((table) => table.name),
+    });
 
     return {
-      microservices: promptResultObj.microservices
-        .sort((microservice) => -1 * microservice.tables.length)
-        .map((microservice) => ({
-          name: microservice.name,
-          functionality: microservice.functionality,
-          tables: microservice.tables
-            .filter((dataModelName) => {
-              const isDuplicatedAlreadyUsed =
-                usedDuplicatedEntities.has(dataModelName);
-              if (duplicatedEntities.has(dataModelName)) {
-                usedDuplicatedEntities.add(dataModelName);
-              }
-              return (
-                originalResourceEntitiesSet.has(dataModelName) &&
-                !isDuplicatedAlreadyUsed
-              );
-            })
-            .map((dataModelName) => {
-              const entityNameIdMap = originalResource.entities.reduce(
-                (map, entity) => {
-                  map[entity.name] = entity;
-                  return map;
-                },
-                {} as Record<string, EntityDataForBtm>
-              );
-
-              return {
-                name: dataModelName,
-                originalEntityId: entityNameIdMap[dataModelName]?.id,
-              };
-            }),
-        }))
-        .filter((microservice) => microservice.tables.length > 0),
+      microservices: promptResultObj.microservices.map((microservice) => ({
+        name: microservice.name,
+        functionality: microservice.functionality,
+        tables: microservice.tables.map((tableName) => {
+          const entityNameIdMap = originalResource.entities.reduce(
+            (map, entity) => {
+              map[entity.name] = entity;
+              return map;
+            },
+            {} as Record<string, EntityDataForBtm>
+          );
+
+          return {
+            name: tableName,
+            originalEntityId: entityNameIdMap[tableName]?.id,
```

Comment:
I wonder if the '?' here is needed and if it's in fact confusing since I believe we expect that *every* table will exist in the map if there are not bugs, right?